pkg/controller/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -142,6 +148,30 @@ func (r *ReconcileMasterUserRecord) addFinalizer(logger logr.Logger, mur *toolch
 	return nil
 }
 
+// check all tiers in use, and make sure that there's a label with a hash of the templates for each of them.
```

Comment:
+1!
This is an important migration step.

```suggestion
// ensureTierTemplatesLabel checks all tiers in use, and make sure that there's a label with a hash of the templates for each of them.
```